src/libraries/System.DirectoryServices/src/CompatibilitySuppressions.xml

```diff
@@ -1,9 +1,15 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
   <Suppression>
-    <DiagnosticId>CP0002</DiagnosticId>
-    <Target>M:System.DirectoryServices.ActiveDirectory.SyncUpdateCallback.#ctor(System.Object,System.IntPtr)</Target>
-    <Left>lib/net6.0/System.DirectoryServices.dll</Left>
-    <Right>lib/net7.0/System.DirectoryServices.dll</Right>
+    <DiagnosticId>CP0015</DiagnosticId>
+    <Target>T:System.DirectoryServices.DirectoryEntry:[T:System.ComponentModel.TypeConverterAttribute]</Target>
```

Comment:
In this case, a required parameter to the attribute's constructor is a `System.Type`. So you're saying that if a type argument to an attribute is determined to be an internal type, we don't emit a diagnostic? That makes sense.

Reply:
I put up a PR for this here: https://github.com/dotnet/sdk/pull/27648.

Reply:
Based on the discussion in https://github.com/dotnet/sdk/pull/27648 it sounds like we treat the omission of the `TypeConverterAttribute` in the contract as a compat difference. Presumably we want to add it to `DirectoryEntry` in the contract and add the internal type that is part of the attribute to the contract as well.

Does that sound right @ericstj?

Reply:
Ah great. I assume that this compat difference is emitted because APICompat isn't able to resolve the typeof() parameter as it points to the internal type. @smasher164 is that right? What should we do about this?